src/test/scala/pt/tecnico/dsi/neutron/SubnetsSpec.scala

```diff
@@ -1,27 +1,36 @@
 package pt.tecnico.dsi.neutron
 
-import cats.effect.IO
+import cats.effect.{IO, Resource}
+import cats.implicits._
+import org.scalatest.Assertion
 import org.scalatest.OptionValues._
-import pt.tecnico.dsi.neutron.models.{Network, Subnet}
-import pt.tecnico.dsi.neutron.services.BulkCreate
+import pt.tecnico.dsi.neutron.models.Subnet
+import pt.tecnico.dsi.neutron.services.{BulkCreate, CrudService}
+import pt.tecnico.dsi.openstack.common.models.WithId
 
-class SubnetsSpec extends CrudSpec[Subnet]("subnet", _.subnets) with BulkCreateSpec[Subnet] {
+class SubnetsSpec extends CrudSpec[Subnet]("subnet") with BulkCreateSpec[Subnet] { self =>
 
+  val service: CrudService[IO, Subnet] with BulkCreate[IO, Subnet] = client.subnets
   val bulkService: NeutronClient[IO] => BulkCreate[IO, Subnet] = _.subnets
-  val updateStub: Subnet.Update = Subnet.Update(name = Some("cool-port"))
-  val createStub: IO[Subnet.Create] = for {
-    neutron <- client
-    net <- neutron.networks.create(Network.Create())
-  } yield Subnet.Create(name = Some("hello"), networkId = net.id, cidr = "192.168.199.0/24", ipVersion = 4)
+  val updateStub: IO[Subnet.Update] = IO { Subnet.Update(name = Some("cool-port")) }
 
-  "Subnets service" should {
-    "update" in {
-      for {
-        neutron <- client
-        stub <- createStub
-        port <- neutron.subnets.create(stub)
-        updated <- neutron.subnets.update(port.id, updateStub)
-      } yield updated.name shouldBe updateStub.name.value
+  override def updateComparator(read: Subnet#Read, update: Subnet#Update): Assertion =
+    read.name shouldBe update.name.value
+
+  override val withStubCreated: Resource[IO, WithId[Subnet.Read]] =
+    withNetworkCreated.flatMap { network =>
+      val create = service.create {
+        Subnet.Create(name = Some("hello"), networkId = network.id, cidr = "192.168.199.0/24", ipVersion = 4)
```

Comment:
To avoid collisions you should use a random name. Copy the random name funcions from here https://github.com/ist-dsi/scala-designate-client/blob/master/src/test/scala/pt/tecnico/dsi/openstack/designate/Utils.scala#L52 to Utils. And use  `withRandomName` to generate a random name for the Subnet.